source/fuzz/fact_manager/data_synonym_and_id_equation_facts.h

```diff
@@ -20,6 +20,7 @@
 
 #include "source/fuzz/data_descriptor.h"
 #include "source/fuzz/equivalence_relation.h"
+#include "source/fuzz/fact_manager/irrelevant_value_facts.h"
```

Comment:
I wonder whether a forward reference might be more appropriate here (with the #include happening in the .cpp file).

The #include works because we only do things in one direction, but if in the future we needed consistency checks in the other direction then we would have to fall back to forward referencing.

Reply:
In fact, we do do it both ways round and you have forward-referenced (necessarily) the other way.  I think it would be better to keep it symmetric and forward reference in both cases.

Reply:
I have now used a forward reference in both cases